### ISSUE TITLE
[Runtime Bug]: Unreal engine gets stuck resizing window

### ISSUE THREAD

**Issue by mmdanggg2** [NVIDIAGameWorks/rtx-remix#583]
### Describe the bug

When resizing the game window it can often get stuck in a loop resizing back and forth between the previous and new size/resolution.

### How do you reproduce the bug?

Run an Unreal engine game with a dx9 renderer, such as [D3D9Drv](https://www.cwdohnal.com/utglr/) or [D3D9DrvRTX](https://github.com/mmdanggg2/D3D9DrvRTX)
Get it into windowed mode and drag to resize the window around.

### Version

0.5.1+c099b404

### Logs

[NvRemixBridge.log](https://github.com/user-attachments/files/16617947/NvRemixBridge.log)
[d3d9.log](https://github.com/user-attachments/files/16617949/d3d9.log)
[UnrealTournament_d3d9.log](https://github.com/user-attachments/files/16617950/UnrealTournament_d3d9.log)

### Crash dumps

[NvRemixBridge.exe_20240814_194453.dmp](https://github.com/user-attachments/files/16617952/NvRemixBridge.exe_20240814_194453.dmp)

### Media

https://github.com/user-attachments/assets/6f18e7ab-9419-4991-8cd2-fbe2751e6fc9

**Comment by NV-LL** [comment]
REMIX-3448 for tracking

**Comment by lvengesanam** [comment]
Few changes were added to fix issues similar to this last month.

@mmdanggg2 Could you test with bridge-remix binary from https://github.com/NVIDIAGameWorks/bridge-remix/actions/runs/10013926678 to verify if the issue persists.

**Comment by mmdanggg2** [comment]
Yes, that does seem to be ok with that version. Although after resizing for a little bit memory usage (RAM+VRAM) steadily increases and it becomes more and more laggy/unresponsive.

**Comment by lvengesanam** [comment]
Could you test to verify if it causes a crash with dxvk-remix binaries from https://github.com/NVIDIAGameWorks/dxvk-remix/actions/runs/10426232995 which resolves issues with recreation of device and swapchain with change in resolution?

Also, your fix under - https://github.com/NVIDIAGameWorks/dxvk-remix/pull/77 was with respect to this issue?

**Comment by mmdanggg2** [comment]
Just tried it with those versions of dxvk and bridge. Still crashing after resizing a few times:

https://github.com/user-attachments/assets/9de4e99b-5f57-4d66-a11e-2db52c4baf59



[NvRemixBridge.log](https://github.com/user-attachments/files/16664427/NvRemixBridge.log)
[d3d9.log](https://github.com/user-attachments/files/16664425/d3d9.log)
[UnrealTournament_d3d9.log](https://github.com/user-attachments/files/16664426/UnrealTournament_d3d9.log)


The crash when minimising from fullscreen seems to also be fixed but now introduced issues with double cursor and I'm unable to interact with the menus properly afterwards:

https://github.com/user-attachments/assets/a30dbe99-e9da-4e95-8a36-6d7aa40d9b2e

**Comment by lvengesanam** [comment]
Thanks for validating with updated binaries.

With respect to the crash on resizing:
Have noticed this issue several months ago on certain system with stock dxvk too wherein several rapid changes to resolution were made.  Could you confirm if this occurs with stock dxvk ?

Will get back to you on the double cursor issue.

**Comment by mmdanggg2** [comment]
Had a go with dxvk 2.4 and it was happily scaling nice and smoothly for a while until I got bored so, safe to say not an issue there.

**Comment by lvengesanam** [comment]
Thanks for confirming. We are still looking into this issue.

With respect to double cursor seen, could you try setting 'client.DirectInput.forward.mousePolicy' to '0' in 'bridge.conf' to verify if it fixes the issue?

**Comment by lvengesanam** [comment]
Thanks for all the tests. Double cursor issue with minimizing seems to be a new one, could you please file a separate bug for that.

We will be looking at both the issues.

**Comment by mmdanggg2** [comment]
Submitted that here: #591 
Seems like the `client.DirectInput.forward.mousePolicy = 0` option actually made no difference.

**Comment by lvengesanam** [comment]
Thanks for creating a separate bug.

To confirm the newer bridge version resolves the double cursor issue prior to minimizing right?

**Comment by mmdanggg2** [comment]
That was not a problem before either, with 0.5.4, it works fine but crashes when minimising.

**Comment by lvengesanam** [comment]
Thanks for the confirmation. We will be looking into this.

**Comment by lvengesanam** [comment]
@mmdanggg2 We have been unable to repro this issue anymore this TOT binaries.
Could you check if the issue is resolved for you with latest binaries?

**Comment by mmdanggg2** [comment]
Yea, just tried with the latest action builds, can't seem to get it stuck looping like this now. But after resizing a few times it maxes out my VRAM and stops rendering, only producing a white flash.

**Comment by lvengesanam** [comment]
Thanks for testing!

We haven't been able to repro this particular issue even after several tries.
Could you update the bug with latest logs along with system information where you noticed this issue?

We will look further into this.

**Comment by mmdanggg2** [comment]
[d3d9.log](https://github.com/user-attachments/files/17688351/d3d9.log)
[remix-dxvk.log](https://github.com/user-attachments/files/17688353/remix-dxvk.log)
[NvRemixBridge.log](https://github.com/user-attachments/files/17688354/NvRemixBridge.log)

https://github.com/user-attachments/assets/c7a56786-8645-4762-9654-480a453f330b
